jwst/combine_1d/combine1d.py

```diff
@@ -619,64 +618,53 @@ def check_exptime(exptime_key):
     """Combine the input spectra.
 
     Parameters
     ----------
-    asn_file : str
-        The association file name.
+    input_model : `~jwst.datamodels.DataModel`
+        The input spectra.  This will likely be a ModelContainer object.
 
     exptime_key : str
         A string identifying which keyword to use to get the exposure time,
-        which is used as a weight when combining spectra.
-
-    interpolation : str
-        The type of interpolation between input pixels.
+        which is used as a weight when combining spectra.  The value should
+        be one of:  "exposure_time" (the default), "integration_time",
+        or "unit_weight".
 
     background : bool, default=False
         If the flux data are actually background rather than a target
         spectrum, `background` should be set to True.  In this case, the
         values read from the flux column of each input spectrum will be
         divided by the npixels column (if that column exists).  This is
         to convert the values to background per pixel.
+
+    Returns
+    -------
+    output_model : `~jwst.datamodels.DataModel`
+        A datamodels.CombinedSpecModel object.
     """
 
     log.debug("Using exptime_key = {}.".format(exptime_key))
     if background:
         log.debug("The FLUX data will be treated as background data.")
 
-    with open(asn_file) as asn_file_handle:
-        asn = load_asn(asn_file_handle)
-
-    # Get the name to use for the output file.
-    output_name = asn["products"][0]["name"]
-    # Get the dictionary that contains the input file names.
-    input_files = asn["products"][0]["members"]
-    del asn
-
     exptime_key = check_exptime(exptime_key)
 
     input_spectra = []
-    for file_dict in input_files:
-        file = file_dict["expname"]
-        ms = datamodels.open(file)
-        try:
-            dummy = len(ms.spec)
-            del dummy
-        except AttributeError:
-            log.error("Input file {} is a {}; skipping ..."
-                      .format(file, type(ms)))
-            ms.close()
-            continue
-        for in_spec in ms.spec:
-            input_spectra.append(InputSpectrumModel(ms, in_spec, exptime_key,
-                                                    background))
-        ms.close()
+    if isinstance(input_model, datamodels.ModelContainer):
+        for ms in input_model:
+            for in_spec in ms.spec:
+                input_spectra.append(InputSpectrumModel(
+                                ms, in_spec, exptime_key, background))
+    else:
+        for in_spec in input_model.spec:
+            input_spectra.append(InputSpectrumModel(
+                                input_model, in_spec, exptime_key, background))
 
     output_spec = OutputSpectrumModel()
     output_spec.assign_wavelengths(input_spectra)
-    output_spec.accumulate_sums(input_spectra, interpolation)
+    output_spec.accumulate_sums(input_spectra, "nearest")
```

Comment:
Technically the interpolation method used in `combine_1d` is `linear`, not `nearest neighbour`.

Reply:
Linear interpolation is used in `to_pixel`, yes, but the `interpolation` parameter was passed to `accumulate_sums` instead, where it referred to taking the nearest pixel in the output file.  I left this parameter in `spec` for the time being to avoid problems with user's copies of combine_1d.cfg that would include this parameter, but I'll remove the parameter completely if you like.  Most references to `interpolation` in the code have been removed.